proof/crefine/ARM/Arch_C.thy

```diff
@@ -3962,10 +3954,10 @@ lemma Arch_decodeInvocation_ccorres:
                   elim!: ccap_relationE split: if_split_asm)
    apply (clarsimp split: list.split)
   apply (clarsimp simp: cap_lift_asid_pool_cap cap_lift_page_directory_cap
-                        cap_to_H_def to_bool_def
-                        cap_page_directory_cap_lift_def
+                        cap_to_H_def cap_page_directory_cap_lift_def to_bool_def
                  elim!: ccap_relationE split: if_split_asm)
   done
```

Comment:
Adding `and_one_neq_simps` did break a few proofs. It's a bit hard to diagnose what is going wrong, but as an example, in this case I think that it is due to the proof resulting in a state roughly saying
```to_bool (x && 1) ⟹ x && 1 ≠ 0```
Previously, the proof solved this by simplifying to
```⟦to_bool (x && 1); x && 1 = 1⟧ ⟹ False```
which was easily solvable with existing rules about `to_bool`.
However, now `and_one_neq_simps` is used to rewrite the goal to
```to_bool (x && 1) ⟹ x && 1 = 1```
which requires unfolding `to_bool_def` to solve.

I think I'm ok with the limited breakages that adding `and_one_neq_simps` to the simpset has caused, but I'd also be happy to rework it if people disagree. Suggestions are also welcome if anyone has ideas that could improve the automation around these constants and rules

Reply:
Well, that's somewhat unexpected. I think your observation of the breakage leans me towards also saying this breakage is ok. It wasn't the to_bool simps that were doing the heavy lifting here, it was the \noteq 0 plus clarsimp. Once that rule applies, there's not much else you can do other than unfold `to_bool`.